x/interchainquery/keeper/keeper.go

```diff
@@ -67,3 +71,20 @@ func (k *Keeper) SubmitICQRequest(ctx sdk.Context, query types.Query, forceUniqu
 
 	return nil
 }
+
+// Re-submit an ICQ, generally used after a timeout
+func (k *Keeper) RetryICQRequest(ctx sdk.Context, query types.Query) error {
```

Comment:
Seems like this code is mostly a copy paste from `SubmitICQRequest`, should we pull it out of both fns into a helper? I personally think it is more clear as written, just a bit redundant

Reply:
I had that same debate as well, but there's actually 1 big difference - SubmitICQRequest generates the query ID, whereas retry uses the same query ID

Reply:
ah good point, makes sense to keep as is then